disco/disco_test.go

```diff
@@ -277,6 +278,35 @@ func TestDiscover(t *testing.T) {
 			t.Errorf("discovered.services not nil (empty); should be")
 		}
 	})
+	t.Run("discovery error", func(t *testing.T) {
+		// Make a channel and then ignore messages to simulate a Client.Timeout
+		donec := make(chan bool, 1)
+		portStr, close := testServer(func(w http.ResponseWriter, r *http.Request) {
+			<-donec
+		})
+		defer close()
+		defer func() { donec <- true }()
+
+		givenHost := "localhost" + portStr
+		host, err := svchost.ForComparison(givenHost)
+		if err != nil {
+			t.Fatalf("test server hostname is invalid: %s", err)
+		}
+
+		d := New()
```

Comment:
You can make this run really fast by temporarily setting a very low client timeout. Without this, the test eats the entire timeout (11 sec)

```suggestion
		d := New()
		
		transport := d.Transport.(*http.Transport)

		origTimeout := transport.ResponseHeaderTimeout
		transport.ResponseHeaderTimeout = 10 * time.Millisecond
		defer func() { transport.ResponseHeaderTimeout = origTimeout }()
```